src/Adapter/Product/CommandHandler/SetProductCustomizationFieldsHandler.php

```diff
@@ -0,0 +1,190 @@
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\CommandHandler;
+
+use CustomizationField;
+use PrestaShop\PrestaShop\Adapter\Product\AbstractCustomizationFieldHandler;
+use PrestaShop\PrestaShop\Adapter\Product\CustomizationFieldPersister;
+use PrestaShop\PrestaShop\Adapter\Product\CustomizationFieldProvider;
+use PrestaShop\PrestaShop\Adapter\Product\ProductUpdater;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\Command\SetProductCustomizationFieldsCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\CommandHandler\AddCustomizationFieldHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\CommandHandler\SetProductCustomizationFieldsHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\CustomizationField as CustomizationFieldDTO;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\CustomizationFieldDeleterInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\ValueObject\CustomizationFieldId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+
+/**
+ * Handles @see  SetProductCustomizationFieldsCommand using legacy object model
+ */
+class SetProductCustomizationFieldsHandler extends AbstractCustomizationFieldHandler implements SetProductCustomizationFieldsHandlerInterface
+{
+    /**
+     * @var AddCustomizationFieldHandlerInterface
+     */
+    private $addCustomizationFieldHandler;
+
+    /**
+     * @var CustomizationFieldDeleterInterface
+     */
+    private $customizationFieldDeleter;
+
+    /**
+     * @var CustomizationFieldPersister
+     */
+    private $customizationFieldPersister;
+
+    /**
+     * @var CustomizationFieldProvider
+     */
+    private $customizationFieldProvider;
+
+    /**
+     * @var ProductUpdater
+     */
+    private $productUpdater;
+
+    /**
+     * @param AddCustomizationFieldHandlerInterface $addCustomizationFieldHandler
+     * @param CustomizationFieldPersister $customizationFieldPersister
+     * @param CustomizationFieldDeleterInterface $customizationFieldDeleter
+     * @param CustomizationFieldProvider $customizationFieldProvider
+     * @param ProductUpdater $productUpdater
+     */
+    public function __construct(
+        AddCustomizationFieldHandlerInterface $addCustomizationFieldHandler,
+        CustomizationFieldPersister $customizationFieldPersister,
+        CustomizationFieldDeleterInterface $customizationFieldDeleter,
+        CustomizationFieldProvider $customizationFieldProvider,
+        ProductUpdater $productUpdater
+    ) {
+        $this->addCustomizationFieldHandler = $addCustomizationFieldHandler;
```

Comment:
This one is not needed any more, the persister manages it all It's great that we can remove these sub handlers dependencies